src/parallel_bush.c

```diff
@@ -119,21 +119,21 @@ void updateBushB_par(int origin, network_type *network, bushes_type *bushes,
         }
     }
 
-//    /* If strict criterion fails, try a looser one */
-//    if (newArcs == 0) {
-//        for (ij = 0; ij < network->numArcs; ij++) {
-//            i = network->arcs[ij].tail;
-//            j = network->arcs[ij].head;
-//            if (bushes->LPcost_par[origin][i]==-INFINITY && bushes->LPcost_par[origin][j]>-INFINITY)
-//                continue; /* No path to extend */
-//            if (bushes->flow_par[origin][ij] == 0 && bushes->LPcost_par[origin][i] < bushes->LPcost_par[origin][j]
-//                && (network->arcs[ij].tail == origin2node(network, origin)
-//                    || network->arcs[ij].tail >= network->firstThroughNode))
-//            {
-//                bushes->flow_par[origin][ij] = NEW_LINK;
-//            }
-//        }
-//    }
+    /* If strict criterion fails, try a looser one */
+    if (newArcs == 0 && parameters->updateBushScanType == LONGEST_BUSH_PATH) {
```

Comment:
Will this always work as a definitive check? Dr. Boyles mentioned "LONGEST_USED_OR_SP thing was me trying to prevent this kind of bug.... ". Maybe we should double check this and make sure that there are no edge cases.

Reply:
I think this makes sense because this criterion can only be applied if we use longest_bush_path for the path labels not longest used path. see the full_log.txt that I sent in the conversation.

Reply:
This bug appeared in the extreme case of broken links with relatively high costs kept in for connectivity requirements and while Can was running the thousands of configurations of TAP.